roles/openstack/neutron/tasks/cleanup.yml

```diff
@@ -0,0 +1,30 @@
+---
+# This file is used to cleanup OpenStack neutron resources
+
+- name: Cleanup routers
+  vars:
+      prefix: "{{ provisioner.prefix | default(uuid) }}-"
+  os_router:
+      interfaces: "{{ item.value.attach_subnets | map('add_prefix', prefix) | join(',') }}"
+      name: "{{ prefix }}-{{ item.value.name | default(router_name) }}"
+      network: "{{ item.value.network | default(router_network) }}"
+      state: absent
+  with_dict: "{{ provisioner.neutron.routers }}"
+  when: provisioner.neutron.routers is defined
+  ignore_errors: yes
+
+- name: Cleanup subnets
+  os_subnet:
+      name: "{{ provisioner.prefix | default(uuid) }}-{{ item.value.name }}"
+      state: absent
+  with_dict: "{{ provisioner.neutron.subnets }}"
+  when: provisioner.neutron.subnets is defined
+  ignore_errors: yes
```

Comment:
If we ignore errors, we need to reflect that cleanup failed some how.

my suggestion:
register each task (subnet, router, etc...) and run some sort of:

```
fail:
    msg: cleanup failed. Please review output to make sure no resources are leaking
when: subnets|failed or router|failed or ...
```